src/hotspot/share/gc/shenandoah/shenandoahCollectorPolicy.hpp

```diff
@@ -94,7 +94,15 @@ class ShenandoahCollectorPolicy : public CHeapObj<mtGC> {
     return _consecutive_degenerated_gcs;
   }
 
+  static bool is_allocation_failure(GCCause::Cause cause) {
+    return cause == GCCause::_allocation_failure
+        || cause == GCCause::_shenandoah_allocation_failure_evac
+        || cause == GCCause::_shenandoah_humongous_allocation_failure;
+  }
```

Comment:
Would it make sense to move this implementation also to the .cpp file like the other static `is_...` methods below?

Reply:
Or is this guaranteeing inlining into the caller's body, which you might prefer for the callers?